src/matmul.jl

```diff
@@ -1147,12 +1148,12 @@ function _generic_matmatmul_nonadjtrans!(C, A, B, alpha, beta)
         for j in axes(C, 2)
             B_1j = B[b1, j]
             for i in axes(C, 1)
-                C_ij = A[i, a1] * B_1j
+                C_ij = @stable_muladdmul MulAddMul(alpha, false)(A[i, a1] * B_1j)
                 C[i,j] = zero(C_ij + C_ij)
             end
         end
     end
-    iszero(alpha) && return C
+    (iszero(alpha) || isempty(A) || isempty(B)) && return C
```

Comment:
I think this is safe, given that we've already dealt with `beta!=0` above.

Reply:
Yes, in that part it does `C .*= beta`. If `iszero(alpha)`, then this is all that needs to be done in

```julia
C = A*B*alpha + C*beta
```

This change is unrelated, it just avoids "running empty loops" in the sequel.